torch/distributed/rendezvous.py

```diff
@@ -131,8 +136,6 @@ def _error(msg):
     def _env_error(var):
         return _error("environment variable %s expected, but not set" % var)
 
-    if not url.startswith("env://"):
```

Comment:
What's the reason to remove this?

Reply:
Remove a unnecessary assert
urllib.parse.urlunparse(parts)
Construct a URL from a tuple as returned by urlparse(). The parts argument can be any six-item iterable. This may result in a slightly different, but equivalent URL, if the URL that was parsed originally had unnecessary delimiters (for example, a ? with an empty query; the RFC states that these are equivalent).
For example:
env:// with rank=1
After urlunparse, the format will be
env?rank=1

Reply:
1. Both tcp and file doesn't have such assertion
2. It is what I explained above.